types/parameterized_types.go

```diff
@@ -0,0 +1,101 @@
+package types
+
+import (
+	"fmt"
+)
+
+// IntegerParam represents a single integer parameter for a parameterized type
+// Example: VARCHAR(L1) -> L1 is the integer parameter
+type IntegerParam struct {
```

Comment:
why is this a struct as opposed to a type alias of string?

Reply:
fixed it